core/src/main/java/org/elasticsearch/cluster/routing/AllocationId.java

```diff
@@ -198,6 +201,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     public static AllocationId fromXContent(XContentParser parser) throws IOException {
-        return ALLOCATION_ID_PARSER.parse(parser, new AllocationId.Builder()).build();
+        return ALLOCATION_ID_PARSER.parse(parser, new AllocationId.Builder(), () -> ParseFieldMatcher.STRICT).build();
```

Comment:
I figured that as well. this seems something very internal that we only generate and parse, so I tend to agree. although having these cases make these constants needed besides tests, which means that they will be misused like they are now. we can discuss this again later when we try to address this problem more closely, not directly related to this PR.